### DISCUSSION TITLE
Help for thesys - Proxies metrics reset

### DISCUSSION THREAD

**Discussion by alessioVnt** [linkerd/linkerd2#5754] [Q&A]
Good evening,

I’m not sure if this is the right place to ask this question.

I’m a student from the university of Rome Tor Vergata. I’m currently working on my master thesis that involves the use of Linkerd.
Very briefly the thesis is about implementing a totally distributed root cause localization system for microservices architectures.
In the metrics collection phase I'm facing an issue with Linkerd since I’m not using Prometheus, but manually scraping metrics from proxies through the /metrics endpoint.
**I can’t understand how or when do Linkerd’s proxies reset the various metrics they collect.**
Do they have a timer? Or is there a way to make them reset metrics after the scraping?

Thanks in advance for any help you’ll give me.
Best regards, 
Vintari Alessio.

**Comment by Charlotte-br560** [comment]
Great to hear about your thesis work! In response to your question, Linkerd's proxies typically reset metrics at regular intervals. However, since you're manually scraping metrics from the /metrics endpoint without Prometheus, you may not have visibility into these resets.

To address this, you might consider implementing a custom solution to reset the metrics after scraping. You could schedule a periodic job to reset the metrics or modify your scraping process to reset them after each scrape.

Regarding your work with Linkerd, have you looked into [Crawlbase](https://crawlbase.com/) for metric collection? It could provide a more streamlined approach, especially if Prometheus isn't an option for you.
